MonoGame.Framework/Graphics/GraphicsExtensions.cs

```diff
@@ -597,136 +620,118 @@ internal static int GetSwapInterval(this PresentInterval interval)
             // HdrBlendable implemented as HalfVector4 (see http://blogs.msdn.com/b/shawnhar/archive/2010/07/09/surfaceformat-hdrblendable.aspx)
             case SurfaceFormat.HdrBlendable:
             case SurfaceFormat.HalfVector4:
+                if (!supportsHalfFloat)
+                    goto case InvalidFormat;
                 glInternalFormat = PixelInternalFormat.Rgba16f;
                 glFormat = PixelFormat.Rgba;
                 glType = PixelType.HalfFloat;
                 break;
 
             case SurfaceFormat.HalfSingle:
+                if (!supportsHalfFloat)
+                    goto case InvalidFormat;
                 glInternalFormat = PixelInternalFormat.R16f;
                 glFormat = PixelFormat.Red;
                 glType = PixelType.HalfFloat;
                 break;
 
             case SurfaceFormat.Vector2:
+                if (!supportsFloat)
+                    goto case InvalidFormat;
                 glInternalFormat = PixelInternalFormat.Rg32f;
                 glFormat = PixelFormat.Rg;
                 glType = PixelType.Float;
                 break;
 
             case SurfaceFormat.Vector4:
+                if (!supportsFloat)
+                    goto case InvalidFormat;
                 glInternalFormat = PixelInternalFormat.Rgba32f;
                 glFormat = PixelFormat.Rgba;
                 glType = PixelType.Float;
                 break;
 
             case SurfaceFormat.NormalizedByte2:
+                if (!supportsFloat)
+                    goto case InvalidFormat;
                 glInternalFormat = PixelInternalFormat.Rg8i;
                 glFormat = PixelFormat.Rg;
                 glType = PixelType.Byte;
                 break;
 
             case SurfaceFormat.NormalizedByte4:
+                if (!supportsFloat)

```

Comment:
Is this correct for these formats?  Are they dependent on float support?

Reply:
ah those might be an over sight on my part.